pkg/store/bucket.go

```diff
@@ -945,12 +949,19 @@ func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Ag
 			if err != nil {
 				return err
 			}
-			out.Counter = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b}
+			out.Counter = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b, Hash: hashChunk(b, calculateChecksum)}
 		}
 	}
 	return nil
 }
 
+func hashChunk(b []byte, doHash bool) uint64 {
+	if !doHash {
+		return 0
+	}
+	return xxhash.Sum64(b)
```

Comment:
We could probably have a sync pool for `xxhash` and reuse `xxhash.New()`.

Reply:
Can you check if I used it correctly? TBH, I have never used it before, but I checked examples in the codebase. Thx for the review.